### PR TITLE
merging and unmerging endpoints & repair server crash

### PR DESCRIPTION
**Pull request by ddnowicki**

### CODE REVIEW

File: src/api/controller/MyController.hpp [hunk 1]
```diff
@@ -78,31 +65,31 @@ class MyController : public oatpp::web::server::api::ApiController {
...
-    } else {
+    if (!json && !json->username && !json->password) {
       responseDto->success = false;
       return createDtoResponse(Status::CODE_400, responseDto);
     }
+
+    std::string token = auth.authenticateUser(json->username, json->password);
+
+    bool loginSuccess = false;
+    if (token != "") loginSuccess = true;
```
Comment (marcinwozniak0) [inline src/api/controller/MyController.hpp]: use { }

File: src/api/controller/MyController.hpp [hunk 2]
```diff
@@ -78,31 +65,31 @@ class MyController : public oatpp::web::server::api::ApiController {
...
-      } else {
-        return createDtoResponse(Status::CODE_403, responseDto);
-      }
-    } else {
+    if (!json && !json->username && !json->password) {
       responseDto->success = false;
       return createDtoResponse(Status::CODE_400, responseDto);
     }
+
+    std::string token = auth.authenticateUser(json->username, json->password);
```
Comment (marcinwozniak0) [inline src/api/controller/MyController.hpp]: const auto token

File: src/api/controller/MyController.hpp (hunk 1 above)
Comment (marcinwozniak0) [reply src/api/controller/MyController.hpp]: the comment applies to all changes, I don't want to add this comment in every place

File: src/api/controller/MyController.hpp [hunk 3]
```diff
@@ -112,32 +99,31 @@ class MyController : public oatpp::web::server::api::ApiController {
...
-        return createDtoResponse(Status::CODE_409, responseDto);
-      }
-    } else {
+    if (!json && !json->username && !json->password && !json->email) {
       responseDto->success = false;
       return createDtoResponse(Status::CODE_400, responseDto);
     }
+
+    std::pair<int, std::string> registerPair =
+        auth.registerUser(json->username, json->password, json->email);
```
Comment (marcinwozniak0) [inline src/api/controller/MyController.hpp]: const auto& [errorCode, token] =  auth.registerUser(json->username, json->password, json->email);

File: src/api/controller/MyController.hpp [hunk 4]
```diff
@@ -302,53 +303,58 @@ class MyController : public oatpp::web::server::api::ApiController {
...
+    if (!json || !json->data || !json->receiptName) {
+      responseDto->success = false;
+      return createDtoResponse(Status::CODE_400, responseDto);
+    }
 
-      if (friends.empty()) {
-        responseDto->success = false;
-        return createDtoResponse(Status::CODE_400, responseDto);
+    std::vector<bakcyl::core::Item> items;
+    for (const auto& pair : *json->data) {
```
Comment (marcinwozniak0) [inline src/api/controller/MyController.hpp]: for (const auto& [idDontKnowWhatIsUnderFirstItem_createApropriateVaraibleName, itemList] : json->data)

File: src/api/controller/MyController.hpp (hunk 1 above)
Comment (ddnowicki) [reply src/api/controller/MyController.hpp]: the commits 20cf9ae and 1f8ab40 in #90 solved the issue

File: src/api/controller/MyController.hpp (hunk 2 above)
Comment (ddnowicki) [reply src/api/controller/MyController.hpp]: the commit 20cf9ae in #90 solved the issue

File: src/api/controller/MyController.hpp (hunk 3 above)
Comment (ddnowicki) [reply src/api/controller/MyController.hpp]: the commit 1e8d812 in #90 solved the issue

File: src/api/controller/MyController.hpp (hunk 4 above)
Comment (ddnowicki) [reply src/api/controller/MyController.hpp]: impossible to do because the oat++ library does not allow it

**Review by marcinwozniak0 — APPROVED** [review]

**Pull request merged**